File: solutions/alfi.compitest/source_gen/alfi/compitest/ActivityBinaryExpressionsTest.cpp
#include "ActivityBinaryExpressionsTest.h"
#include <alf/library/BasicInputOutput.hpp>
#include <string>
#include <alf/library/primitivebehaviors/BooleanFunctions.hpp>

void BinaryExpressionsTest()
{
  // Arithmetic Expressions
  alf::library::BasicInputOutput::WriteLine(std::to_string(12 + 45));
  alf::library::BasicInputOutput::WriteLine(std::to_string(-33));
  alf::library::BasicInputOutput::WriteLine(std::to_string(12 * 45));
  alf::library::BasicInputOutput::WriteLine(std::to_string(121 % 5));

  // Shift Expressions
  alf::library::BasicInputOutput::WriteLine(std::to_string(static_cast<int>((121 << 2))));
  alf::library::BasicInputOutput::WriteLine(std::to_string(static_cast<int>((121 >> 2))));
  alf::library::BasicInputOutput::WriteLine(std::to_string(static_cast<int>((static_cast<int>(static_cast<unsigned>(121) >> 2)))));
  alf::library::BasicInputOutput::WriteLine(std::to_string(static_cast<int>((-121 >> 2))));
  alf::library::BasicInputOutput::WriteLine(std::to_string(static_cast<int>((static_cast<int>(static_cast<unsigned>(-121) >> 2)))));

  // Relational Expressions
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(5 < 9));
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(5 < 5));
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(5 < 3));

  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(5 <= 9));
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(5 <= 5));
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(5 <= 3));

  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(5 > 9));
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(5 > 5));
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(5 > 3));

  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(5 >= 9));
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(5 >= 5));
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(5 >= 3));

  // Equality Expressions
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(1 == 1));
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(1 == 3));
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(std::string("a") == std::string("a")));
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(std::string("a") == std::string("ab")));

  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(1 != 1));
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(1 != 3));
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(std::string("a") != std::string("a")));
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(std::string("a") != std::string("ab")));

  // Logical Expression
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(true & true));
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(true & false));
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(false & false));

  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(true ^ true));
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(true ^ false));
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(false ^ false));

  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(true | true));
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(true | false));
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(false | false));

  // Conditional Expression
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(true && true));
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(true && false));
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(false && false));

  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(true || true));
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(true || false));
  alf::library::BasicInputOutput::WriteLine(alf::library::primitivebehaviors::BooleanFunctions::ToString(false || false));

  // Conditional Test Expression
  alf::library::BasicInputOutput::WriteLine(true ? std::string("val1") : std::string("val2"));
  alf::library::BasicInputOutput::WriteLine(false ? std::string("val1") : std::string("val2"));
  alf::library::BasicInputOutput::WriteLine(true ? (true ? std::string("val1") : std::string("val2")) : (true ? std::string("val3") : std::string("val4")));
  alf::library::BasicInputOutput::WriteLine(false ? (true ? std::string("val1") : std::string("val2")) : (true ? std::string("val3") : std::string("val4")));
}

File: solutions/alfi.compitest/source_gen/alfi/compitest/ActivityBinaryExpressionsTest.cs
/// <filename>
///     ActivityBinaryExpressionsTest.cs
/// </filename>
using Alf.Library;
using Alf.Library.PrimitiveBehaviors;

public class ActivityBinaryExpressionsTest
{
    public static void Execute()
    {
        // Arithmetic Expressions
        BasicInputOutput.WriteLine(IntegerFunctions.ToString(12 + 45));
        BasicInputOutput.WriteLine(IntegerFunctions.ToString(-33));
        BasicInputOutput.WriteLine(IntegerFunctions.ToString(12 * 45));
        BasicInputOutput.WriteLine(IntegerFunctions.ToString(121 % 5));

        // Shift Expressions
        BasicInputOutput.WriteLine(IntegerFunctions.ToString((int) (121 << 2)));
        BasicInputOutput.WriteLine(IntegerFunctions.ToString((int) (121 >> 2)));
        BasicInputOutput.WriteLine(IntegerFunctions.ToString((int) ((int) (unchecked((uint) 121) >> 2))));
        BasicInputOutput.WriteLine(IntegerFunctions.ToString((int) (-121 >> 2)));
        BasicInputOutput.WriteLine(IntegerFunctions.ToString((int) ((int) (unchecked((uint) -121) >> 2))));

        // Relational Expressions
        BasicInputOutput.WriteLine(BooleanFunctions.ToString(5 < 9));
        BasicInputOutput.WriteLine(BooleanFunctions.ToString(5 < 5));
        BasicInputOutput.WriteLine(BooleanFunctions.ToString(5 < 3));

        BasicInputOutput.WriteLine(BooleanFunctions.ToString(5 <= 9));
        BasicInputOutput.WriteLine(BooleanFunctions.ToString(5 <= 5));
        BasicInputOutput.WriteLine(BooleanFunctions.ToString(5 <= 3));

        BasicInputOutput.WriteLine(BooleanFunctions.ToString(5 > 9));
        BasicInputOutput.WriteLine(BooleanFunctions.ToString(5 > 5));
        BasicInputOutput.WriteLine(BooleanFunctions.ToString(5 > 3));

        BasicInputOutput.WriteLine(BooleanFunctions.ToString(5 >= 9));
        BasicInputOutput.WriteLine(BooleanFunctions.ToString(5 >= 5));
        BasicInputOutput.WriteLine(BooleanFunctions.ToString(5 >= 3));

        // Equality Expressions
        BasicInputOutput.WriteLine(BooleanFunctions.ToString(1 == 1));
        BasicInputOutput.WriteLine(BooleanFunctions.ToString(1 == 3));
        BasicInputOutput.WriteLine(BooleanFunctions.ToString("a" == "a"));
        BasicInputOutput.WriteLine(BooleanFunctions.ToString("a" == "ab"));

        BasicInputOutput.WriteLine(BooleanFunctions.ToString(1 != 1));
        BasicInputOutput.WriteLine(BooleanFunctions.ToString(1 != 3));
        BasicInputOutput.WriteLine(BooleanFunctions.ToString("a" != "a"));
        BasicInputOutput.WriteLine(BooleanFunctions.ToString("a" != "ab"));

        // Logical Expression
        BasicInputOutput.WriteLine(BooleanFunctions.ToString(true && true));
        BasicInputOutput.WriteLine(BooleanFunctions.ToString(true && false));
        BasicInputOutput.WriteLine(BooleanFunctions.ToString(false && false));

        BasicInputOutput.WriteLine(BooleanFunctions.ToString(true ^ true));
        BasicInputOutput.WriteLine(BooleanFunctions.ToString(true ^ false));
        BasicInputOutput.WriteLine(BooleanFunctions.ToString(false ^ false));

        BasicInputOutput.WriteLine(BooleanFunctions.ToString(true | true));
        BasicInputOutput.WriteLine(BooleanFunctions.ToString(true | false));
        BasicInputOutput.WriteLine(BooleanFunctions.ToString(false | false));

        // Conditional Expression
        BasicInputOutput.WriteLine(BooleanFunctions.ToString(true && true));
        BasicInputOutput.WriteLine(BooleanFunctions.ToString(true && false));
        BasicInputOutput.WriteLine(BooleanFunctions.ToString(false && false));

        BasicInputOutput.WriteLine(BooleanFunctions.ToString(true || true));
        BasicInputOutput.WriteLine(BooleanFunctions.ToString(true || false));
        BasicInputOutput.WriteLine(BooleanFunctions.ToString(false || false));

        // Conditional Test Expression
        BasicInputOutput.WriteLine(true ? "val1" : "val2");
        BasicInputOutput.WriteLine(false ? "val1" : "val2");
        BasicInputOutput.WriteLine(true ? (true ? "val1" : "val2") : (true ? "val3" : "val4"));
        BasicInputOutput.WriteLine(false ? (true ? "val1" : "val2") : (true ? "val3" : "val4"));
    }

    public static void Main(string[] args)
    {
        ActivityBinaryExpressionsTest.Execute();
    }
}
